src/server/editorServices.ts

```diff
@@ -933,7 +933,16 @@ function createWatchFactoryHostUsingWatchEvents(service: ProjectService, canUseW
             recursive ? watchedDirectoriesRecursive : watchedDirectories,
             path,
             callback,
-            id => ({ eventName: CreateDirectoryWatcherEvent, data: { id, path, recursive: !!recursive } }),
+            id => ({
+                eventName: CreateDirectoryWatcherEvent,
+                data: {
+                    id,
+                    path,
+                    recursive: !!recursive,
+                    // Special case node_modules as we watch it for changes to closed script infos as well
+                    ignoreUpdate: !path.endsWith("/node_modules") ? true : undefined,

```

Comment:
Can this path be `\node_modules` on Windows?

Reply:
Slightly related: I did notice another watch request for `node_modules/@types`, so I wonder if these could also `ignoreUpdate`?

Reply:
Hm, I think I misunderstood our discussion around changes in `node_modules` then. I had assumed TS only cares about added/deleted events in `node_modules` to rerun module resolution, but not for updates. So now with your change, folder watchers ignore updates unless its `node_modules` being watched?

Reply:
Yes. we only care about `updates` inside `node_modules` and no other folder.

Reply:
Reference to code here https://github.com/microsoft/TypeScript/blob/main/src/server/editorServices.ts#L3102 this normally works out because with fs.watch most of the time we get "rename" event instead of "change" when doing `npm i` but with parcel you get `update` so that cannot be ignored from vscode watcher usage perspective

Reply:
Yeah I understand. My idea was to collect events to `node_modules` in the TS extension and then send you a single event for the module. That can be a `added` event to emulate the `rename` you got from `fs.watch`. Its a bit of a hack and very specific to this case but would essentially reduce the event load on `node_modules`, which can be a huge source of file events.

Reply:
What i meant by `ignoreUpdate` here is that we really dont care if there is file that was modified within that folder and thats true for any folder watched except `node_modules`. So from protocol wise this change looks good. 

On whether to send single `directory event` for `module` inside `node_modules` or all files can be totally customized irrespective of this protocol change. In general, the more specific you are about the `path` the better reuse happens between programs. We were anyways getting these events from `fs.watch` and we handle both cases since the fileName may or may not be present when the event comes through `fs.watch`.